src/datajudge/requirements.py

```diff
@@ -592,6 +593,7 @@ def add_varchar_regex_constraint(
         relative_tolerance: float = 0.0,
         aggregated: bool = True,
         n_counterexamples: int = 5,
+        computation_in_db=False,
```

Comment:
I suggest to make this two separate `add_...` functions instead. Maybe call one "db" and the other "py" and deprecate the current name.

Reply:
Unfortunately we cannot deprecate the in-memory version since there is no sqlalchemy implementation to do regex matching for mssql.

Aside from that I also wondered whether to switch with different methods or via parameters and was unsure. Not trying to oppose you suggestion: How do you see that trade-off?

Reply:
Hard to come up with a general rule. Indicators that something should be a different function/method/program:

- Different interface: Python-style regexes vs. database specific regexes
- Behavior is substantially different: Moved work (= CPU and memory resources used) from Python to database.
- Parameter combinations that are not applicable: `allow_none` is not applicable to the database version.